### PR TITLE
Migrated more cmd code

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: common/credentialFactory.go [hunk 1]
```diff
@@ -166,6 +173,120 @@ func CreateBlobClientFromServiceClient(blobURLParts blob.URLParts, client *blobs
 	return blobClient, nil
 }
 
+func CreateContainerClient(blobURL string, credInfo *CredentialInfo, options *container.ClientOptions, credOpOptions *CredentialOpOptions) (*container.Client, error) {
```
Comment (gapra-msft) [inline common/credentialFactory.go:176]: Ideally I don't want so much CreateXXXClient code. I will follow up this PR with some code to clean this up a bit and parse the URL, get service client, then get the appropriate container/blob client so we isolate this SAS/OAuth/SharedKey switch statement to one method

File: cmd/benchmark.go [hunk 2]
```diff
@@ -186,12 +186,15 @@ func (raw rawBenchmarkCmdArgs) appendVirtualDir(target, virtualDir string) (stri
...
+		p, err := blob.ParseURL(target)
+		if err != nil {
+			return "", fmt.Errorf("error parsing the url %s. Failed with error %s", target, err.Error())
+		}
 		if p.ContainerName == "" || p.BlobName != "" {
 			return "", errors.New("the blob target must be a container")
 		}
 		p.BlobName = virtualDir
-		result = p.URL()
+		return result.String(), err
```
Comment (vibhansa-msft) [inline cmd/benchmark.go:197]: 'result' is not used in blob.ParseURL(), is it correct variable to return?

Comment (gapra-msft) [reply cmd/benchmark.go]: fixed

**Review by vibhansa-msft — APPROVED** [review]

**Review by siminsavani-msft — APPROVED** [review]

**Review by tasherif-msft — APPROVED** [review]

File: cmd/copyEnumeratorInit.go [hunk 3]
```diff
@@ -460,29 +462,23 @@ func (cca *CookedCopyCmdArgs) createDstContainer(containerName string, dstWithSA
...
+		bsc, err := common.CreateBlobServiceClient(accountRoot, dstCredInfo, nil, options)
 		if err != nil {
 			return err
 		}
+		bcc := bsc.NewContainerClient(containerName)
 
-		bsu := azblob.NewServiceURL(*dstURL, dstPipeline)
-		bcu := bsu.NewContainerURL(containerName)
-		_, err = bcu.GetProperties(ctx, azblob.LeaseAccessConditions{})
+		_, err = bcc.GetProperties(ctx, nil)
```
Comment (nakulkar-msft) [inline cmd/copyEnumeratorInit.go:471]: We'll save a call to service if we skip this. We already check ignore ContainerAlreadyExists in after the Create call.

**Review by adreed-msft — APPROVED** [review]

File: cmd/copyEnumeratorInit.go [hunk 4]
```diff
@@ -460,29 +462,23 @@ func (cca *CookedCopyCmdArgs) createDstContainer(containerName string, dstWithSA
 			return err
 		}
 
-		dstURL, err := url.Parse(accountRoot)
-
+		bsc, err := common.CreateBlobServiceClient(accountRoot, dstCredInfo, nil, options)
```
[the remark this answers is not in the document]
Comment (gapra-msft) [reply cmd/copyEnumeratorInit.go:465]: added to doc as cleanup item

**Pull request merged**